src/backend/access/external/pxfutils.c

```diff
@@ -47,6 +52,24 @@ void port_to_str(char **port, int new_port)
 }
 
 /*
+ * get_hdfs_location_from_filespace
+ *
+ * Get hdfs location from pg_filespace_entry
+ * The returned path needs to be pfreed by the caller.
+ */
+void get_hdfs_location_from_filespace(char** path)
+{
+	Assert(NULL != path);
```

Comment:
This code gets the file space of the current DB (MyDatabaseId), so even if a specific table is configured to a different file space it shouldn't affect it.
Do you think it would be safer to get the file space of DB 'template1' instead of the current session's DB? I followed the examples in other places in the code to get the file space, and they either use the current table's DB or MyDatabaseId.